mission/mission_core/missions/ScanTheCode_mission.py

```diff
@@ -0,0 +1,249 @@
+        And can be accessed like so:
+            camera_data.frame
+            camera_data.results
+
+        PositionData is a class that contains the current position and heading of the vehicle.
+        And can be accessed like so:
+            position_data.lat : float
+            position_data.lon : float
+            position_data.position : tuple (lat, lon)
+            position_data.heading : float
+
+        The output perc_cmd has the following format (does not have to be filled):
+        {
+            "start": ["camera_name"],
+            "stop": ["camera_name"],
+            "record": ["camera_name"],
+            "stop_record": ["camera_name"],
+            "load_model": [("camera_name", "model_path")],
+            "stop_model": ["camera_name"]
+        }
+
+        The output gnc_cmd has the following format (does not have to be filled):
+        {
+            "heading": float, # The target heading for the boat to turn towards
+            "vector": tuple, # The target vector for the boat to head towards (will override heading)
+            "poshold": bool, # Whether to set the GNC to poshold mode (heading is not maintained but position is)
+            "waypoint": tuple, # The target waypoint to set the GNC to (sending any other commands will override this)
+            "end_mission": bool # Whether to end the mission
+        }
+
+        '''
+        
+        # Run one yolo inference on the provided frame at camera_data.frame
+        self.storageArray = camera_data.results[0]
+        
+        if(len(self.storageArray) != 0):
+            if(self.debug):
+                print("Interim processing")
+                print(self.storageArray)
+            returnVal = filter(self.storageArray)
+            records = self.analyze(returnVal)
+
+            self.colors, self.pattern = self.results(records)
+            if(self.debug):
+                print("printing results")
+                print(self.colors[0] + self.colors[1] + self.colors[2])
+                print("printing dict outside")
+                print(records)
+            
+            max_vote = records.get(self.pattern)
+            if max_vote is None:
+                max_vote = 0
+            # if(max_vote > 0):
+            #     # Want to update colors for heartbeat
+            #     update_colors_on_server(colors, max_vote, len(storageArray))
+            # else:
+            #     # No Colors to update, can just say black
+            #     update_colors_on_server(["Black", "Black", "Black"], 0, len(storageArray))
+        
+        
+        
+        
+        perc_cmd = {
+            "start": ["center"],
+            "stop": ["port", "starboard"],
+            "record": ["center"],
+            "stop_record": ["port", "starboard"],
+            "load_model": [("center", "ML_Resources/stcA.pt")],
+            "stop_model": ["port", "starboard"]
+        }
+        gnc_cmd = {"poshold": True}
```

Comment:
So the one downside with poshold is that it maintains position, but not heading (as of now). So it may be worth to instead of poshold do something where you make the boat turn (using target heading) to point towards the tower. So like if the tower is in the right of the frame (you can figure where it is using the bbox coordinates) you can tell the boat to turn with target_heading = curr_heading + 5 or something until its in the center etc and vice versa if its on the left. Essentially a shitty P controller that should do the job. 

Alternatively, you can turn on all the cameras and run the model on all of them and do the entire process for all the cams and you should have a very large FOV.